tests/integration_test/run_tournament.py

```diff
@@ -0,0 +1,155 @@
+# Util imports
+from input_output.file_output import load_graph_list_from_filepath, create_csv_file, write_csv_line
+from input_output.sys_output import title, passed
+import tkinter as tk
+from tkinter import filedialog
+from time import time
+# Algorithm imports
+from tests.integration_test.settings import *
+from tests.integration_test.isomorphism_problem import are_isomorph, amount_of_isomorphisms
+
+"""
+General integration test for the Graph Isomorphisms problem.
+Settings of the algorithm must be changed in > settings.py <
+"""
+
+"""
+FILE INPUT
+"""
+# Select files to evaluate
+root = tk.Tk()
+root.withdraw()
+file_paths = filedialog.askopenfilenames()
+
+"""
+RUN TOURNAMENT
+"""
+for filepath in file_paths:
+    graphs = load_graph_list_from_filepath(filepath)
+    filename = (filepath.split("/")[-1]).split(".")[0]
+
+    passed("Starting evaluating " + filename)
+
+    isomorphisms = []       # Data structure that saves all isomorphic pairs (or more than 2, if that is the case)
+    iso_count = {}          # Data structure that saves for each graph the amount of automorphisms
+    total_time = 0
+    # In this first loop, for each combination, it is determined if they are isomorphic or not
+    for i in range(len(graphs) - 1):
+        for j in range(i + 1, len(graphs)):
+            G = graphs[i]
+            H = graphs[j]
+
+            G_copy = G.copy()
+            H_copy = H.copy()
+            start_isomorph = time()
+            are_isomorph_actual = are_isomorph(G_copy, H_copy)
+            end_isomorph = time()
+            total_time += round((end_isomorph - start_isomorph), 3)
+
+            # Only save results if the combination of graphs is isomorphic, unless you have to know the automorphisms
+            # of each graph, then save each graph in de isomorphisms data structure.
+            if are_isomorph_actual:
+                # If one of the two graphs is already in a isomorphic pair, the other graph belongs to it too
+                already_in_results = -1
+                for pair in isomorphisms:
+                    if i in pair:
+                        pair.append(j)
+                        already_in_results = i
+                        break
+                    elif j in pair:
+                        pair.append(i)
+                        already_in_results = j
+                        break
+                # Otherwise a new pair of isomorphisms should be added to the list
+                if already_in_results < 0:
```

Comment:
Stel nu dat graaf 1 en 2 van de lijst isomorf zijn en graaf 1 en 4 ook. Misschien kunnen we dan toevoegen dat we voor de combinatie 2 en 4 niet meer uitrekenen of ze isomorf zijn, omdat we dat al weten.

Reply:
Ja slim, toegevoegd!